aopy/analysis/behavior.py

```diff
@@ -454,4 +456,50 @@ def vector_angle(vector, in_degrees=False):
     if in_degrees:
         angle = angle*180/np.pi
 
-    return angle
\ No newline at end of file
+    return angle
+
+
+def correlate_trajectories(trajectories, center=True, verbose=False):
+    '''
+    Correlates multiple trajectory datasets across trials by computing the 
+    Pearson correlation coefficient (R) between all pairs of trials. This function computs R for each trajectory dimension, then returns a weighted average based on the variance. 
+
+    Args:
+        trajectories (nt, ntrials, ndim): Trajectories to correlate, 
+                                 where `nt` is the number of timepoints, 
+                                 `ntrials` is the number of trials, and 
+                                 `ndims` is the number of dimensions for each trajectory (i.e. x, y, z).
+        center (bool): If each trial should be centered before computing correlation. (Safaie et al. 2023 sets this to true)
+        verbose (bool): If `True`, prints a progress bar during computation via the tqdm module. 
+
+    Returns:
+        ndarray: A 2D numpy array of R^2 scores between each pair of trials. 
+                 The shape of the output will be (ntrials, ntrials).
+    '''
+
+    nt, ntrials, ndims = trajectories.shape
```

Comment:
it seems a bit strange to have the input be (nt, ntr, 3) instead of (nt, 3, ntr) since the individual trials are all (nt, 3). it would seem easier to stack trials rather than have to inserting a trial dimension in between the time and ch dimensions